### ISSUE TITLE
Syncing fork since commit history edit.

### ISSUE THREAD

**Issue by jnellis** [coding-horror/basic-computer-games#703]
So I noticed my repo was borked from the recent git history mangling, effectively disconnecting it from easily fetching upstream merges. Roughly 1000 forks are currently like this so its a bit of an issue perhaps (I'm no git wizard so maybe its not.)  After some searching about I came across these incantations to force your local repo to fetch from this upstream repo then force push to your github fork.
 
https://gist.github.com/glennblock/1974465

**Comment by MartinThoma** [comment]
The simplest way to ensure you are not running into issues is to delete the fork + fork again.

**Comment by jnellis** [comment]
Yep. Didn't want to do that though, thought there might be some other way. There's at least one fork that has been forked itself and if anyone was using **Issues** on their fork it would be lost I think. Anyway, I didn't see this addressed anywhere and things like this can have roadblocking effect on further participation.  Again, not a git wizard, but I'm betting there is still some dark magic that could have been invoked to delete specific large commits without disrupting the flow of things too much.

**Comment by mojoaxel** [comment]
As @jnellis mentioned the easiest way is to reset the local branch:

If not done already you first have to add the `coding-horror` repository as a remote:
```sh
git remote add upstream https://github.com/coding-horror/basic-computer-games.git
```

After that you need to checkout `master` and overwrite it with the remote:
```sh
git fetch upstream
git checkout master
git reset --hard upstream/master
```
